tasks/main.yml

```diff
@@ -0,0 +1,117 @@
+---
+
+- name: check if passwords are specified
+  assert:
+    that:
+      - database_password is truthy
+      - database_root_password is truthy
+
+- name: install MariaDB on RedHat Systems
+  dnf:
+    name:
+      - mariadb-server
+      - mariadb
+      - python3-PyMySQL
+
+- name: create systemd configuration directory
+  file:
+    path: /etc/systemd/system/mariadb.service.d/
+    state: directory
+    mode: '0755'
+
+- name: limit number of open files
+  copy:
+    content: |
+      [Service]
+      LimitNOFILE=10000
+    dest: /etc/systemd/system/mariadb.service.d/limits.conf
+    mode: '0644'
+  register: reload_systemd_daemon
+  notify: restart MariaDB
+
+- name: reload systemd daemon  # noqa no-handler
+  systemd:
+    daemon_reload: true
+  when: reload_systemd_daemon is changed
+
+- name: configure mariadb
+  ini_file:
+    dest: /etc/my.cnf.d/opencast.cnf
+    section: mysqld
+    option: '{{ item.key }}'
+    value: '{{ item.value }}'
+    owner: root
+    group: root
+    mode: '0644'
+    state: present
+  loop:
+    - key: max_connections
+      value: '512'
+    - key: character_set_server
+      value: utf8mb4
+    - key: bind-address
+      value: '*'
+  notify: restart MariaDB
+
+- name: start MariaDB
+  service:
+    name: mariadb
+    state: started
+    enabled: true
+
+- name: wait for MariaDB port to become available
+  wait_for:
+    port: 3306
+    delay: 1
+
+- name: set MariaDB root user password
+  mysql_user:
+    name: '{{ database_root_user }}'
+    password: '{{ database_root_password }}'
+    login_user: '{{ database_root_user }}'
+    login_password: '{{ database_root_password }}'
```

Comment:
Do we need this? This should never be correct if we actually change the password.

Reply:
> We need this in case we want to change the root password later.

How would that work? You would update `database_root_password` to the new password you want to set which is also the one this task is trying to log in with and since it's then the incorrect current password, that would fail, wouldn't it?

Reply:
…maybe it does need it for checking the passwrd.
Not sure about that.

Reply:
It actually does need it for checking after the first run. If not this task fails.